cpp/Esri/ArcGISRuntime/Toolkit/CoordinateConversionController.h

```diff
@@ -0,0 +1,309 @@
+   */
+  GenericListModel* coordinateFormats() const;
+
+  /*!
+   * \brief List of textual representations of the current point in different 
+   * formats.
+   * Internally, this is a GenericListModel with an elementType of
+   * CoordinateConversionResult.
+   * 
+   * \sa GenericListModel
+   * 
+   * \return GenericListModel of CoordinateConversionResult elements.
+   */
+  GenericListModel* conversionResults() const;
+
+  /*!
+   * \brief Returns the current point.
+   * 
+   * The current point represents the one point that all elements within
+   * the conversionResults list-model are tied to.
+   * 
+   * \return Point 
+   */
+  Point currentPoint() const;
+
+  /*!
+   * \brief Converts the current point held by this controller as a 2D point 
+   * relative to the current window.
+   * 
+   * \return The point returned by currentPoint as a screen coordinate. 
+   */
+  Q_INVOKABLE QPointF screenCoordinate() const;
+
+  /*!
+   * \brief Distance between camera and current point when zoomToCurrentPoint
+   * is invoked.
+   * 
+   * If geoView is a SceneView, then this value represents the distance
+   * between the camera and the point returned by currentPoint when 
+   * zoomToCurrentPoint is called.
+   * 
+   * If geoView is a MapView, this value has no effect on zoom calculations.
+   * 
+   * The distance is in the units of the GeoView's SpatialReference.
+   * 
+   * Defaults to 1500.
+   * 
+   * \return distance between current point and viewpoint camera.
+   */
+  double zoomToDistance() const;
+
+  /*!
+   * \brief Set the zoomTodistance.
+   * \sa zoomToDistance
+   * \param distance distance value in the units of the GeoView's
+   * SpatialReference.
+   */
+  void setZoomToDistance(double distance);
+
+  /*!
+   * \brief Returns whether this controller is actively listening for mouse
+   * events.
+   * 
+   * When listening for mouse events, the Controller will update the
+   * current point to whatever point is returned by a click event on the
+   * GeoView.
+   * 
+   * \return true This controller is listening to mouse events on the GeoView
+   * and will actively update the CurrentPoint.
+   * \return false This controller is not listening to mouse events on the
+   * GeoView.
+   */
+  bool inPickingMode() const;
+
+  /*!
+   * \brief Set whether the controller should actively be listening to mouse
+   * events on the GeoView.
+   * \sa inPickingMode
+   * \param mode True for listening to mouse events, false for ignoring mouse
+   * events.
+   */
+  void setInPickingMode(bool mode);
+
+signals:
+  /*! \brief Emitted when the geoView has changed. */
+  void geoViewChanged();
+
+  /*! \brief Emitted when the currentPoint has changed. */
+  void currentPointChanged(const Point& point);
+
+  /*!
+   * \brief Emitted when the currentPoint property has changed.
+   * \note This signal is a QVariant containing the Point cast to a Geometry
+   * type. Due to technical limitations a `Point` is not a Qt MetaType
```

Comment:
as discussed, seems like we could register this in the API though?

Reply:
I would like to, all the code saying we do this because of API reasons can be removed by then.